test/suites/integration/ec2nodeclass_kubelet_test.go

```diff
@@ -0,0 +1,142 @@
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/samber/lo"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	karpv1 "sigs.k8s.io/karpenter/pkg/apis/v1"
+	karpv1beta1 "sigs.k8s.io/karpenter/pkg/apis/v1beta1"
+)
+
+var _ = Describe("EC2nodeClass Kubelet Configuration", func() {
+	var betaNodePool *karpv1beta1.NodePool
+	var betaNodeClass *v1beta1.EC2NodeClass
+	BeforeEach(func() {
+		betaNodeClass = test.BetaEC2NodeClass(v1beta1.EC2NodeClass{
+			Spec: v1beta1.EC2NodeClassSpec{
+				AMIFamily: lo.ToPtr(v1beta1.AMIFamilyAL2023),
+				Role:      fmt.Sprintf("KarpenterNodeRole-%s", env.ClusterName),
+				Tags: map[string]string{
+					"testing/cluster": env.ClusterName,
+				},
+				SecurityGroupSelectorTerms: []v1beta1.SecurityGroupSelectorTerm{
+					{
+						Tags: map[string]string{"karpenter.sh/discovery": env.ClusterName},
+					},
+				},
+				SubnetSelectorTerms: []v1beta1.SubnetSelectorTerm{
+					{
+						Tags: map[string]string{"karpenter.sh/discovery": env.ClusterName},
+					},
+				},
+			},
+		})
+		betaNodePool = &karpv1beta1.NodePool{
+			ObjectMeta: metav1.ObjectMeta{Name: strings.ToLower(randomdata.SillyName())},
+			Spec: karpv1beta1.NodePoolSpec{
+				Template: karpv1beta1.NodeClaimTemplate{
+					Spec: karpv1beta1.NodeClaimSpec{
+						NodeClassRef: &karpv1beta1.NodeClassReference{
+							APIVersion: object.GVK(betaNodeClass).GroupVersion().String(),
+							Kind:       object.GVK(betaNodeClass).Kind,
+							Name:       betaNodeClass.Name,
+						},
+						Requirements: []karpv1beta1.NodeSelectorRequirementWithMinValues{
+							{
+								NodeSelectorRequirement: corev1.NodeSelectorRequirement{
+									Key:      karpv1beta1.CapacityTypeLabelKey,
+									Operator: corev1.NodeSelectorOpExists,
+								},
+							},
+						},
+						Kubelet: &karpv1beta1.KubeletConfiguration{MaxPods: lo.ToPtr[int32](20)},
+					},
+				},
+			},
+		}
+	})
+	It("should expect v1beta1 nodePool to have kubelet compatibility annotation", func() {
+		env.ExpectCreated(betaNodeClass, betaNodePool)
+		Eventually(func(g Gomega) {
+			np := &karpv1.NodePool{}
+			err := env.Client.Get(env, client.ObjectKeyFromObject(betaNodePool), np)
+			g.Expect(err).ToNot(HaveOccurred())
+
+			hash, found := np.Annotations[karpv1.KubeletCompatabilityAnnotationKey]
+			g.Expect(found).To(BeTrue())
+			g.Expect(hash).To(Equal(np.Hash()))
+		})
+	})
+	It("should expect nodeClaim to not drift when kubelet configuration on v1 nodeClass is same as v1beta1 nodePool", func() {
+		pod := coretest.Pod()
+		env.ExpectCreated(pod, betaNodeClass, betaNodePool)
+		env.EventuallyExpectHealthy(pod)
+		nodeClaim := env.EventuallyExpectCreatedNodeClaimCount("==", 1)[0]
+		_, found := nodeClaim.Annotations[v1.AnnotationKubeletCompatibilityHash]
+		Expect(found).To(BeTrue())
+
+		np := &karpv1.NodePool{}
+		err := env.Client.Get(env, client.ObjectKeyFromObject(betaNodePool), np)
```

Comment:
Can we use `env.ExpectExists`?

Reply:
https://github.com/aws/karpenter-provider-aws/blob/c32ccc4e38420428afe99ff03e8560b2c5857649/test/suites/drift/suite_test.go#L779